conn.go

```diff
@@ -325,10 +326,53 @@ func (c *Conn) Subprotocol() string {
 	return c.subprotocol
 }
 
-// Close closes the underlying network connection without sending or waiting
-// for a close message.
-func (c *Conn) Close() error {
-	return c.conn.Close()
+// Close sends close frame and waits for one in response
+// it expects two args. `closeCode int` and `closeMessage string` in order
+// it uses variadic args to maintain backwards compatibility
+func (c *Conn) Close(args ...interface{}) error {
```

Comment:
This is an incompatible change.   It also removes the ability to simply close the network connection. I'd leave this function as is.

Reply:
The way I see it, it has a pro and a con.
Pro would be same code would now do proper connection close.
Con would be if someone was using close as means to close underlying connection, it's no more a case. We can export new function to close underlying conn though.

Reply:
The Close function in the PR breaks the following:

- The change in function signature can cause compilation errors.
- The change in semantics will break applications that depend on Close simply closing the network connections.  This includes applications that close the connection to handle an abusive client and applications that implement the closing handshake already.